src/ray/id.cc

```diff
@@ -22,16 +26,17 @@ UniqueID UniqueID::from_binary(const std::string &binary) {
 
 const UniqueID UniqueID::nil() {
   UniqueID result;
-  std::fill_n(result.id_, kUniqueIDSize, 255);
+  uint8_t *data = result.mutable_data();
+  std::fill_n(data, kUniqueIDSize, 255);
   return result;
 }
 
-bool is_nil(const UniqueID &rhs) {
+bool UniqueID::is_nil() const {
   int i = kUniqueIDSize;
-  const uint8_t *data = rhs.data();
-  while (--i > 0 && data[i] == 255) {
+  const uint8_t *d = data();
+  while (--i > 0 && d[i] == 255) {
```

Comment:
I might be misreading this, but doesn't this fail to check `d[0]`? Also, we don't have any guarantees about the order in which the two boolean clauses are executed, right? But if they are executed out of order then we will have problems.

Reply:
Good catch!